core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1577,7 +1582,7 @@ public ManagedChannel build() {
           // TODO(creamsoup) prevent main channel to shutdown if oob channel is not terminated
           return new ManagedChannelImpl(
                   managedChannelImplBuilder,
-                  transportFactory,
+                  originalTransportFactory,
```

Comment:
Shouldn't this be `oobTransportFactory`? Otherwise there is no `CallCredentialsApplyingTransport` and no callCreds can be sent over oobChannel?

Reply:
I checked the new test and if the test is passing then this change seems to be correct.

Reply:
This is the part that I mentioned was subtle in the description. It needs to be the original factory because this is getting passed to a _new_ `ManagedChannelImpl`. _That new instance_ will wrap the factory. If we use oobTransportFactory here then the CallCredentials are applied twice. This was a pre-existing bug that I only noticed because of the test I added.

Reply:
Thanks for the explanation. Great catch.